src/Server/AwsCloudFormationTemplates.fs

```diff
@@ -12,481 +9,221 @@ open Shared
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
-
-let remapAppScalableTarget
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (resourceContext: AwsResourceContext)
-    (spec: CustomRemapSpecification) 
-    : RemappedSpecResult = 
-    let data = resourceData[resource.logicalId]
-    let physicalIdParts = data["Id"].Split("|")
-    let importId = 
-        if physicalIdParts.Length = 3 then
-            [physicalIdParts[2]; physicalIdParts[0]; physicalIdParts[1]]
-            |> String.concat spec.delimiter
-        else ""
-                
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
-
-let validateSecurityGroupRule
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (spec: CustomRemapSpecification) 
-    : bool =
-    ((resourceData[resource.logicalId]).ContainsKey "IpProtocol") &&
-    ((resourceData[resource.logicalId]).ContainsKey "GroupId")
-
-let filterSecurityGroupRules
+let tryFindSecurityGroupRule
     (data: Dictionary<string,string>)
-    (rules: Map<string,JObject>)
-    : Map<string,JObject> =
+    (rules: Map<string,JObject>) =
     rules
     |> Map.filter (fun id props -> 
         props.Property("GroupId").Value.ToString() = data["GroupId"])
-    |> Map.filter (fun id props ->
+    |> Map.tryFindKey (fun id props ->
         data
         |> Seq.forall (fun entry ->
-            if entry.Key = "Id" || entry.Key = "resourceType" then true
+            if entry.Key = "Id" || entry.Key = "resourceType" then 
+                true
             elif props.ContainsKey entry.Key then
                 let prop = props.Property(entry.Key)
                 prop.Value.ToString() = data[prop.Name]
-            else false
+            else 
+                false
         ))
 
-let remapSecurityGroupIngress
-    (resource: AwsCloudFormationResource)
-    (resourceData: Dictionary<string, Dictionary<string,string>>)
-    (resourceContext: AwsResourceContext)
-    (spec: CustomRemapSpecification) 
-    : RemappedSpecResult =
-    let data = resourceData[resource.logicalId]
-    let filteredRules = filterSecurityGroupRules data resourceContext.securityGroupIngressRules
-    let importId =
-        if not ((Seq.length filteredRules) = 1) then 
-            ""
-        else (Seq.exactlyOne filteredRules).Key
-    
-    let resourceType = spec.pulumiType
-    let logicalId = resource.logicalId.Replace("-", "_")
-    {
-        resourceType = resourceType
-        logicalId = logicalId
-        importId = importId
-    }
+let requireParts (keys: string list) (data: Dictionary<string, string>) =
```

Comment:
I'm thinking we should also check that the value is not ""

Reply:
Yeah that sounds better